core/blockchain.go

```diff
@@ -2388,18 +2388,26 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 				} else {
 					now := currentEpochSuperCommittee.Epoch
 					// only insert if APR for current epoch does not exists
-					if _, ok := stats.APRs[now.Int64()]; !ok {
-						stats.APRs[now.Int64()] = *aprComputed
-						// check and clean aprs for epochs prior to current minus APRHistoryLength
-						nowMinus100 := now.Sub(now, big.NewInt(staking.APRHistoryLength)).Int64()
-						keysToRemove := []int64{}
-						for i := range stats.APRs {
-							if i < nowMinus100 {
-								keysToRemove = append(keysToRemove, i)
+					aprEntry := staking.APREntry{now, *aprComputed}
+					l := len(stats.APRs)
+					if l == 0 {
+						// first time inserting apr for validator
+						stats.APRs = append(stats.APRs, aprEntry)
+					} else {
```

Comment:
rid this index , its caused by just 1 known event and it causes a permentant shift of the code. Prefer not having elses